### ISSUE TITLE
Implement constructor function for alternating layer ansatz

### ISSUE THREAD

**Issue by ducthanh1991** [pasqal-io/qadence#532]
Hello,

Have you considered implementing a function to construct the alternating layer ansatz defined in the papers of [Nakaji et al.]( https://quantum-journal.org/papers/q-2021-04-19-434/) and [Cerezo et al.](https://www.nature.com/articles/s41467-021-21728-w) that would be similar to the existing `hea` function of the module `qadence.constructors.ansatze`?

**Comment by chMoussa** [comment]
Hi @ducthanh1991, we have not considered until you asked. It is probably a matter of changing slightly changing the function yielding the entangler, right?

**Comment by ducthanh1991** [comment]
Hello @chMoussa, 

Yes, I guess that one would need to specify all the same parameters as for the `hea` function plus one additional parameter for the size of the support of the "local" block of the alternating layer ansatz.

something like:

```
ansatz = alt_digital(
        depth=`4,
        n_qubits=12,
        m_qubits_block=4,
        operations=[RX],
        entangler=CNOT,
        )
display(ansatz,size="15,15")
```
would yield:
![image](https://github.com/user-attachments/assets/feaa2755-34bf-499b-a729-19641e03a6c7)

**Comment by chMoussa** [comment]
> Hello @chMoussa,
> 
> Yes, I guess that one would need to specify all the same parameters as for the `hea` function plus one additional parameter for the size of the support of the "local" block in the alternating layer ansatz.
> 
> something like:
> 
> ```
> ansatz = alt_digital(
>         depth=`4,
>         n_qubits=12,
>         m_qubits_block=4,
>         operations=[RX],
>         entangler=CNOT,
>         )
> display(ansatz,size="15,15")
> ```
> 
> would yield: ![image](https://private-user-images.githubusercontent.com/96840542/359603415-feaa2755-34bf-499b-a729-19641e03a6c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyNDI3NjcsIm5iZiI6MTcyNDI0MjQ2NywicGF0aCI6Ii85Njg0MDU0Mi8zNTk2MDM0MTUtZmVhYTI3NTUtMzRiZi00OTliLWE3MjktMTk2NDFlMDNhNmM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIxVDEyMTQyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MzA1MzMzZWY5M2E0M2UyZmNjZGRlOTQxNTcwZGI0NWE3YWQxYjI3ZDQzYmFkZGUxNjc4YmE1OGI3MjBmYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gDWmW51wvs69QY6hjgkuVruETLS-_9qLFCtysSJWfXQ)

Ok I will see what is possible to do with the current implementation, unless you already have something on your side?

**Comment by ducthanh1991** [comment]
Hello @chMoussa, 

Yes I do have something that I wrote. I'll work on it during the weekend and make a pull request.

**Comment by ducthanh1991** [comment]
Hi @chMoussa,

I have created the pull request #565. Could you take a look?

**Comment by chMoussa** [comment]
Closes with #565
